### DISCUSSION TITLE
Output API Payloads from Monaco Dry Runs for Testing & Troubleshooting Templates/Configs

### DISCUSSION THREAD

**Discussion by kahoffman** [Dynatrace/dynatrace-configuration-as-code#1013] [Ideas]
I have been working with go template syntax to try to reduce the number of json template files I have to maintain and to move all of my configuration into yaml. I am trying to figure out how to test what the final json looks like after monaco evaluates the yaml and json template. I have been using a go template test web page for which I have to manipulate the yaml config to work with due to the way parameters are defined. The ideal feature would be a way to output the api call payloads for the calls that will be made as part of the dry run. Maybe specify an output directory to see the results. This would make it easier to validate that json template is evaluating the way you expect and to debug when something goes wrong. Alternatively maybe have a test tool to do the same, i.e. reads the config yaml and template and output the evaluated template json. I could probably write a test tool in go myself but there should be a way to do this already.

**Comment by Laubi** [comment]
Hi, @kahoffman , thanks for opening this discussion. Your suggestions are valid and valuable.
We're already evaluating the required features for a proper `dry-run` internally. 
Currently, this feature has a low priority, so I can't promise that we deliver it in the near future.
Still, we'll start discussions to maybe raise the priority, as this feature gets requested more and more often.
Again, thanks for your input, as your suggested features have not yet been considered.
